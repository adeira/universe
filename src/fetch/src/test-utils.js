// @flow

/* eslint-disable import/no-extraneous-dependencies */

import { setupServer, type Server } from 'msw/node';

export { rest } from 'msw';

export const server: Server = setupServer();
