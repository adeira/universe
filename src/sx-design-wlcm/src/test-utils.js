// @flow

import { init as FbtInit, IntlVariations as FbtIntlVariations } from 'fbt';
import { render as renderWithoutProviders } from '@testing-library/react';

const initFbt = () => {
  FbtInit({
    translations: {}, // TODO
    hooks: {
      getViewerContext: () => ({
        GENDER: FbtIntlVariations.GENDER_UNKNOWN,
        regionalLocale: 'en_US',
        locale: 'en_US',
      }),
    },
  });
};

// export custom render method(s) and other helpers:
export { initFbt, renderWithoutProviders as render };
