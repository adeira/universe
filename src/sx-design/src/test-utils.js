// @flow

import { type Element } from 'react';
import { init as FbtInit, IntlVariations as FbtIntlVariations } from 'fbt';
// eslint-disable-next-line import/no-extraneous-dependencies
import {
  render as renderWithoutProviders,
  type RenderOptionsWithoutCustomQueries,
} from '@testing-library/react';

import getFbtTranslationsForLocale from './getFbtTranslationsForLocale';
import type { SupportedLocales } from './constants';

const initFbt = () => {
  FbtInit({
    translations: getFbtTranslationsForLocale('en-US'),
    hooks: {
      getViewerContext: () => ({
        GENDER: FbtIntlVariations.GENDER_UNKNOWN,
        locale: 'en-US',
      }),
    },
  });
};

const renderWithProviders = (
  ui: Element<any>,
  {
    locale = 'en-US',
    ...renderOptions
  }: { +locale: SupportedLocales, ...RenderOptionsWithoutCustomQueries } = {},
): $FlowFixMe => {
  return renderWithoutProviders(ui, {
    wrapper: ({ children }) => {
      const SxDesignProvider = require('./SxDesignProvider').default;
      return <SxDesignProvider locale={locale}>{children}</SxDesignProvider>;
    },
    ...renderOptions,
  });
};

// re-export everything
// eslint-disable-next-line import/no-extraneous-dependencies
export * from '@testing-library/react';

// override render method
export { initFbt, renderWithProviders as render, renderWithoutProviders };
