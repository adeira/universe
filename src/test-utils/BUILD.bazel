load("//:bazel/macros/build_yarn_workspace.bzl", "build_yarn_workspace")
load("//:bazel/macros/test_yarn_workspace.bzl", "test_yarn_workspace")

build_yarn_workspace(
    name = "test-utils",
    package_name = "@adeira/test-utils",
    deps = [
        "@npm//graphql",
    ],
)

test_yarn_workspace(
    name = "test",
    deps = [
        "@npm//graphql",
    ],
)
