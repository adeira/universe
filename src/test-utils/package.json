{
  "name": "@adeira/test-utils",
  "description": "Set of tools for easier testing in JS.",
  "homepage": "https://github.com/adeira/universe/tree/master/src/test-utils",
  "license": "MIT",
  "private": false,
  "version": "0.6.0",
  "main": "src/index",
  "sideEffects": false,
  "dependencies": {
    "@adeira/js": "^1.3.0",
    "@babel/runtime": "^7.12.5"
  },
  "peerDependencies": {
    "graphql": "^14.6.0"
  }
}
