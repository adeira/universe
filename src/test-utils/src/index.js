// @flow

import generateTestsFromFixtures from './generateTestsFromFixtures';
import evaluateGraphQLResolver from './evaluateGraphQLResolver';

export { evaluateGraphQLResolver, generateTestsFromFixtures };
