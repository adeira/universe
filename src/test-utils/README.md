```text
yarn add @adeira/test-utils
```

Optionally, install TS types (Flow types are included by default):

```text
yarn add --dev @types/adeira__test-utils
```

# `generateTestsFromFixtures`

This function loads all the fixtures (their file content) from your folder and it snapshots the result of your callback operation. This is handy when you have several example files, you want to perform some operation on top of them and snapshot the results.

```js
import { generateTestsFromFixtures } from '@adeira/test-utils';

function operation(input) {
  return doSomethingWithTheInput(input);
}

generateTestsFromFixtures(`${__dirname}/__fixtures__`, operation);
```

There are two additional modes which allow you to control the fixtures behavior more precisely. There are based on the fixture name:

- `simpleExample.graphql` - fixture will snapshot as expected
- `simpleExample.only.graphql` - only these snapshots will run (for debug only)
- `simpleExample.error.graphql` - it's expected that fixture will throw an error (expected to fail tests)

The error file convention is required to prevent unintentional snapshots of exceptions.

# `evaluateGraphQLResolver`

```js
const fields = Location.getFields();
expect(
  evaluateGraphQLResolver(fields.countryFlagURL, {
    country: ' ... ', // test value
  }),
).toBe(' ... ');
```
