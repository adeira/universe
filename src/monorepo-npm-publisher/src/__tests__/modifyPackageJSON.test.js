// @flow

import snapshotDiff from 'snapshot-diff';
import { generateTestsFromFixtures } from '@adeira/test-utils';

import modify from '../modifyPackageJSON';

expect.addSnapshotSerializer(snapshotDiff.getSnapshotDiffSerializer());

generateTestsFromFixtures(`${__dirname}/fixtures`, (rawInput) => {
  const input = JSON.parse(rawInput);
  const modifiedInput = modify(input);
  const changes = snapshotDiff(input, modifiedInput);
  return `${JSON.stringify(modifiedInput, null, 2)}\n~~~~~\n${changes}`;
});
