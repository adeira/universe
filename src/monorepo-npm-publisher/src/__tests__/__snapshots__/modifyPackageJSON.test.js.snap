// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches expected output: missingMainAndModule.json 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
{
  "name": "missingMainAndModule",
  "description": "no changes expected",
  "version": "0.0.0",
  "dependencies": {
    "//": "none"
  }
}

~~~~~~~~~~ OUTPUT ~~~~~~~~~~
{
  "name": "missingMainAndModule",
  "description": "no changes expected",
  "version": "0.0.0",
  "dependencies": {
    "//": "none"
  }
}
~~~~~
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`matches expected output: withMain.json 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
{
  "name": "withMain",
  "description": "module should be added and JS extension from main removed",
  "version": "0.0.0",
  "main": "src/index.js",
  "dependencies": {
    "//": "none"
  }
}

~~~~~~~~~~ OUTPUT ~~~~~~~~~~
{
  "main": "src/index",
  "module": "src/index.mjs",
  "name": "withMain",
  "description": "module should be added and JS extension from main removed",
  "version": "0.0.0",
  "dependencies": {
    "//": "none"
  }
}
~~~~~
Snapshot Diff:
- First value
+ Second value

  Object {
    "dependencies": Object {
      "//": "none",
    },
    "description": "module should be added and JS extension from main removed",
-   "main": "src/index.js",
+   "main": "src/index",
+   "module": "src/index.mjs",
    "name": "withMain",
    "version": "0.0.0",
  }
`;

exports[`matches expected output: withMainAndDisabledModule.json 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
{
  "name": "withMainAndDisabledModule",
  "description": "module should be removed and main should stay as is",
  "version": "0.0.0",
  "main": "src/index.js",
  "module": false,
  "dependencies": {
    "//": "none"
  }
}

~~~~~~~~~~ OUTPUT ~~~~~~~~~~
{
  "main": "src/index.js",
  "name": "withMainAndDisabledModule",
  "description": "module should be removed and main should stay as is",
  "version": "0.0.0",
  "dependencies": {
    "//": "none"
  }
}
~~~~~
Snapshot Diff:
- First value
+ Second value

@@ -2,9 +2,8 @@
    "dependencies": Object {
      "//": "none",
    },
    "description": "module should be removed and main should stay as is",
    "main": "src/index.js",
-   "module": false,
    "name": "withMainAndDisabledModule",
    "version": "0.0.0",
  }
`;

exports[`matches expected output: withMainAndDisabledModuleNoExtension.json 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
{
  "name": "withMainAndDisabledModuleNoExtension",
  "description": "module should be removed and main should stay as is",
  "version": "0.0.0",
  "main": "src/index",
  "module": false,
  "dependencies": {
    "//": "none"
  }
}

~~~~~~~~~~ OUTPUT ~~~~~~~~~~
{
  "main": "src/index",
  "name": "withMainAndDisabledModuleNoExtension",
  "description": "module should be removed and main should stay as is",
  "version": "0.0.0",
  "dependencies": {
    "//": "none"
  }
}
~~~~~
Snapshot Diff:
- First value
+ Second value

@@ -2,9 +2,8 @@
    "dependencies": Object {
      "//": "none",
    },
    "description": "module should be removed and main should stay as is",
    "main": "src/index",
-   "module": false,
    "name": "withMainAndDisabledModuleNoExtension",
    "version": "0.0.0",
  }
`;

exports[`matches expected output: withMainAndModule.json 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
{
  "name": "withMainAndModule",
  "description": "module should stay as is and JS extension should be removed from main",
  "version": "0.0.0",
  "main": "src/index.js",
  "module": "src/index.mjs",
  "dependencies": {
    "//": "none"
  }
}

~~~~~~~~~~ OUTPUT ~~~~~~~~~~
{
  "main": "src/index",
  "module": "src/index.mjs",
  "name": "withMainAndModule",
  "description": "module should stay as is and JS extension should be removed from main",
  "version": "0.0.0",
  "dependencies": {
    "//": "none"
  }
}
~~~~~
Snapshot Diff:
- First value
+ Second value

  Object {
    "dependencies": Object {
      "//": "none",
    },
    "description": "module should stay as is and JS extension should be removed from main",
-   "main": "src/index.js",
+   "main": "src/index",
    "module": "src/index.mjs",
    "name": "withMainAndModule",
    "version": "0.0.0",
  }
`;

exports[`matches expected output: withMainAndModuleNoExtension.json 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
{
  "name": "withMainAndModuleNoExtension",
  "description": "no change expected",
  "version": "0.0.0",
  "main": "src/index",
  "module": "src/index.mjs",
  "dependencies": {
    "//": "none"
  }
}

~~~~~~~~~~ OUTPUT ~~~~~~~~~~
{
  "main": "src/index",
  "module": "src/index.mjs",
  "name": "withMainAndModuleNoExtension",
  "description": "no change expected",
  "version": "0.0.0",
  "dependencies": {
    "//": "none"
  }
}
~~~~~
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`matches expected output: withMainNoExtension.json 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
{
  "name": "withMainNoExtension",
  "description": "module should be added with MJS extension and main should stay as is",
  "version": "0.0.0",
  "main": "src/index",
  "dependencies": {
    "//": "none"
  }
}

~~~~~~~~~~ OUTPUT ~~~~~~~~~~
{
  "main": "src/index",
  "module": "src/index.mjs",
  "name": "withMainNoExtension",
  "description": "module should be added with MJS extension and main should stay as is",
  "version": "0.0.0",
  "dependencies": {
    "//": "none"
  }
}
~~~~~
Snapshot Diff:
- First value
+ Second value

@@ -2,8 +2,9 @@
    "dependencies": Object {
      "//": "none",
    },
    "description": "module should be added with MJS extension and main should stay as is",
    "main": "src/index",
+   "module": "src/index.mjs",
    "name": "withMainNoExtension",
    "version": "0.0.0",
  }
`;

exports[`matches expected output: withModule.json 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
{
  "name": "withModule",
  "description": "module should stay as is and main without JS extension should be added",
  "version": "0.0.0",
  "module": "src/index.mjs",
  "dependencies": {
    "//": "none"
  }
}

~~~~~~~~~~ OUTPUT ~~~~~~~~~~
{
  "main": "src/index",
  "module": "src/index.mjs",
  "name": "withModule",
  "description": "module should stay as is and main without JS extension should be added",
  "version": "0.0.0",
  "dependencies": {
    "//": "none"
  }
}
~~~~~
Snapshot Diff:
- First value
+ Second value

  Object {
    "dependencies": Object {
      "//": "none",
    },
    "description": "module should stay as is and main without JS extension should be added",
+   "main": "src/index",
    "module": "src/index.mjs",
    "name": "withModule",
    "version": "0.0.0",
  }
`;
